imageio/imageio-tiff/src/main/java/com/twelvemonkeys/imageio/plugins/tiff/CCITTFaxDecoderStream.java

```diff
@@ -356,6 +356,9 @@ private int decodeRun(final Tree tree) throws IOException {
             if (n.isLeaf) {
                 total += n.value;
                 if (n.value < 64) {
+                    if (n.value < 0) {
```

Comment:
I didn't want to do this check on the same level. < 64 is a normal flow for terminating codes, while < 0 is an error case. Theoretical this means the inner if is called at least a little less often for long runlengths, where non-terminating codes are used.